gnomad_qc/v3/create_release/make_var_annot_hists.py

```diff
@@ -1,61 +1,164 @@
+        logger.info(
+            "Aggregating hists over ranges determined using first pass run..."
+        )
         hists = ht.aggregate(
             hl.array(
-                [hist_expr.annotate(metric=hist_metric) for hist_metric, hist_expr in hist_ranges_expr.items()]
+                [
+                    hist_expr.annotate(metric=hist_metric)
+                    for hist_metric, hist_expr in hist_ranges_expr.items()
+                ]
             ).extend(
                 hl.array(
                     hl.agg.group_by(
                         create_frequency_bins_expr(
-                            AC=ht.freq[1].AC,
-                            AF=ht.freq[1].AF
+                            AC=ht.freq[1].AC, AF=ht.freq[1].AF
+                        ),
+                        # Decided to use QUALapprox because its formula is easier to interpret than QUAL's
+                        hl.agg.hist(
+                            hl.log10(ht.info.QUALapprox),
+                            *ANNOTATIONS_HISTS["QUALapprox"],
                         ),
-                        hl.agg.hist(hl.log10(ht.info.QUALapprox), 1, 10, 36)
                     )
-                ).map(
-                    lambda x: x[1].annotate(metric=x[0])
-                )
+                ).map(lambda x: x[1].annotate(metric="QUALapprox-"+x[0]))
+            ).extend(
+                hl.array(
+                    hl.agg.group_by(
+                        create_frequency_bins_expr(
+                            AC=ht.freq[1].AC, AF=ht.freq[1].AF
+                        ),
+                        hl.agg.hist(
+                            hl.log10(ht.info.AS_QUALapprox),
+                            *ANNOTATIONS_HISTS["AS_QUALapprox"],
+                        ),
+                    )
+                ).map(lambda x: x[1].annotate(metric="AS_QUALapprox-"+x[0]))
             ),
-            _localize=False
+            _localize=False,
         )
 
-        with hl.hadoop_open(qual_hists_json_path(CURRENT_RELEASE), 'w') as f:
-            f.write(hl.eval(hl.json(hists)))
+        # Defining hist range and bins for allele frequency groups because they needed different ranges
+        inbreeding_bin_ranges = {'under_0.0005': [-0.0005, 0.0005, 50], 'over_0.0005': [-0.25, 1, 50]}
+        ht = ht.annotate(af_bin=create_frequency_bins_expr_inbreeding(AF=ht.freq[1].AF))
+        inbreeding_hists = [
+            ht.aggregate(
+                hl.agg.filter(
+                    ht.af_bin == x,
+                    hl.agg.hist(
+                        ht.info.InbreedingCoeff,
+                        *inbreeding_bin_ranges[x],
+                    )
+                )
+            ).annotate(metric="InbreedingCoeff" + "-" + x) for x in inbreeding_bin_ranges
+        ]
+
+        hists = hl.eval(hl.json(hists))
+        inbreeding_hists = hl.eval(hl.json(inbreeding_hists))
+        hists = hists[:-1] + "," + inbreeding_hists[1:]
```

Comment:
For my own knowledge, why is the last item remove from hists and first from inbreeding_hists?

Reply:
I'm going to mess around with this one again. The reason is I am just removing the `}` from the first and the `{` from the second. I think there is a better way to do this, but at the time of running it I was getting an error because the types of hists (before hl.json) didn't match the type of inbreeding hists and trying to cast to the same type was still giving an error. This was my quick and dirty way of getting the hists quickly because all I really needed was to append the two together.

Reply:
Yeah so the problem I had run into was that inbreeding_hists has this structure `<ArrayStructExpression of type array<struct{bin_edges: array<float64>, bin_freq: array<int32>, n_smaller: int32, n_larger: int32, metric: str}>>`  and hists has `<ArrayStructExpression of type array<struct{bin_edges: array<float64>, bin_freq: array<int64>, n_smaller: int64, n_larger: int64, metric: str}>>` so I can't extend the hists array as is. `ArrayExpression.extend' expects 'a' to be the same type as the caller`

Reply:
Also for my knowledge, what error did it give when you tried to cast one of the hists to the same type as the other? Is the current issue appending int32 vs int64?

Reply:
I've struggled to come up with an alternative here without adding way more code. The python is just so much shorter without casting things. I think this works but maybe just add a comment on what is being removed? Alternatively, join and then just hist.remove('{') and hist.remove('}') ?

Reply:
Sound's good to me, agree I couldn't find a better way currently and this is overall relatively clean and represents what was done in v3.1. I will add a comment